File: games/planetization/bots/java/src/core/api/Response.java
package core.api;

import core.api.commands.Command;
import core.api.commands.SendUnitCommand;
import core.api.commands.SpawnUnitCommand;
import core.api.commands.UnitType;

import java.util.ArrayList;

public class Response {
    public int __uid;
    private ArrayList<Command> commands = new ArrayList<>();

    public void spawnUnit(int planetId, UnitType type)  {
        commands.add(new SpawnUnitCommand(planetId, type));
    }

    public void sendUnit(int unitId, int destinationPlanetId)  {
        commands.add(new SendUnitCommand(unitId, destinationPlanetId));
    }
}

File: games/planetization/bots/kotlin/src/core/api/Response.kt
package core.api

interface Command

class Response {
    var __uid: Int = 0
    private val commands = ArrayList<Command>()

    fun spawnUnit(planetId: Int, type: UnitType) {
        commands.add(SpawnUnitCommand(planetId, type))
    }

    fun sendUnit(unitId: Int, destinationPlanetId: Int) {
        commands.add(SendUnitCommand(unitId, destinationPlanetId))
    }
}

class SendUnitCommand(var unitId: Int, var destinationPlanetId: Int) : Command {
    var __type = this.javaClass.simpleName
}

class SpawnUnitCommand(var planetId: Int, var type: UnitType) : Command {
    var __type = this.javaClass.simpleName
}

enum class UnitType {
    WORKER, WARRIOR
}
